js/src/pages/price-benchmark/price-benchmark-table/index.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { withViewportMatch } from '@wordpress/viewport';
+import { TablePlaceholder } from '@woocommerce/components';
+import { useState, useMemo, useEffect } from '@wordpress/element';
+import { DataViews, filterSortAndPaginate } from '@wordpress/dataviews/wp';
+
+/**
+ * Internal dependencies
+ */
+import './index.scss';
+
+// Defines the base fields (image, title, description) that are consistently used across all tables.
+const BASE_FIELDS = [
+	{
+		id: 'image',
+		enableHiding: false,
+		enableSorting: false,
+		enableGlobalSearch: false,
+		label: __( 'Image', 'google-listings-and-ads' ),
+		render: ( { item } ) => {
+			return (
+				<img
+					src={ item.image }
+					alt={ item.title }
+					className="gla-price-benchmark-table__image"
+				/>
+			);
+		},
+	},
+	{
+		id: 'title',
+		enableHiding: false,
+		enableSorting: false,
+		enableGlobalSearch: true,
+		label: __( 'Product', 'google-listings-and-ads' ),
+	},
+	{
+		id: 'description',
+		enableHiding: false,
+		enableSorting: false,
+		enableGlobalSearch: true,
+		label: __( 'Description', 'google-listings-and-ads' ),
+		render: ( { item } ) => {
+			return <span>{ item.description }</span>;
+		},
+	},
+];
+
+/**
+ * PriceBenchmarkTable component renders a table or other data views for price benchmarking.
+ *
+ * @param {Object} props - The component props.
+ * @param {Array} props.data - The data to be displayed in the table.
+ * @param {Array} props.fields - The fields to be displayed in the table for desktop view.
+ * @param {Array} props.fieldsMobile - The fields to be displayed in the table for mobile view.
+ * @param {boolean} props.isViewportMobile - Indicates if the viewport is in mobile mode.
+ * @param {boolean} props.isReady - Indicates if the data is ready to be displayed.
+ *
+ * @return {JSX.Element} The rendered PriceBenchmarkTable component.
+ */
+const PriceBenchmarkTable = ( {
+	data,
+	fields,
+	fieldsMobile,
+	isViewportMobile,
+	isReady,
+} ) => {
+	const [ view, setView ] = useState( {
+		type: 'table',
+		search: '',
+		page: 1,
+		perPage: 10,
+		layout: {},
+		filters: [],
+		fields: [],
+		titleField: 'title',
+		descriptionField: 'description',
+		mediaField: 'image',
+	} );
+
+	const { data: shownData, paginationInfo } = useMemo( () => {
+		const updatedData = filterSortAndPaginate( data, view, fields );
+
+		return updatedData;
+	}, [ view, data, fields ] );
+
+	const handleOnChangeView = ( newView ) => {
+		setView( newView );
+	};
```

Comment:
Consider memoizing this callback

```
const handleOnChangeView = useCallback((newView) => {
  setView(newView);
}, []);
```

Reply:
Additionally, it doesn't look like this updates the URL structure so that if the page is refreshed you always go back to the first page with default filters applied. For example: go to page 2 and refresh, order by price and refresh, etc.

Reply:
@joemcgill I was not aware we needed to keep table interactions persistent 🤔  Maybe we need a separate ticket for that?

Reply:
I assumed that DataViews handled this out of the box. Given that it doesn't seem to, I think we can do this in a follow-up as a nice to have.

Reply:
I thought so as well 😁 Yeah, let's put it as nice to have. Furthermore, I've updated the callback to use `useCallback`